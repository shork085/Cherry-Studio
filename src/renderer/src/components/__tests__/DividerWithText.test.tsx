import { render, screen } from '@testing-library/react'
import { describe, expect, it } from 'vitest'

import DividerWithText from '../DividerWithText'

describe('DividerWithText', () => {
  it('should render with correct structure and text', () => {
    const text = 'Section Title'
    render(<DividerWithText text={text} />)

    // Verify text is rendered
    const textElement = screen.getByText(text)
    expect(textElement).toBeInTheDocument()
    expect(textElement.tagName).toBe('SPAN')

    // Verify structure
    const dividerContainer = textElement.parentElement as HTMLElement
    expect(dividerContainer).toBeTruthy()
    expect(dividerContainer.tagName).toBe('DIV')
    expect(dividerContainer.children.length).toBe(2)

    // Verify line element exists
    const lineElement = dividerContainer.children[1] as HTMLElement
    expect(lineElement.tagName).toBe('DIV')
  })

  it('should apply custom styles', () => {
    const customStyle = {
      marginTop: '20px',
      marginBottom: '30px',
      padding: '10px'
    }

    const { container } = render(<DividerWithText text="Styled" style={customStyle} />)
    const dividerContainer = container.firstChild as HTMLElement

    expect(dividerContainer).toHaveStyle(customStyle)
  })

  it('should handle edge cases for text prop', () => {
    // Empty string
    const { container, rerender } = render(<DividerWithText text="" />)
    const emptySpan = container.querySelector('span')
    expect(emptySpan).toBeTruthy()
    expect(emptySpan?.textContent).toBe('')

    // Long text
    const longText = 'This is a very long section title that might wrap or cause layout issues'
    rerender(<DividerWithText text={longText} />)
    expect(screen.getByText(longText)).toBeInTheDocument()

    // Special characters
    const specialText = '特殊字符 & Symbols: <>&"\'@#$%'
    rerender(<DividerWithText text={specialText} />)
    expect(screen.getByText(specialText)).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<DividerWithText text="Test Divider" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
