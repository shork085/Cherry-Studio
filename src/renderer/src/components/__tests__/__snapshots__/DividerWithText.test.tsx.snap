// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DividerWithText > should match snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
  margin: 0px 0;
}

.c1 {
  font-size: 12px;
  color: var(--color-text-2);
  margin-right: 8px;
}

.c2 {
  flex: 1;
  height: 1px;
  background-color: var(--color-border);
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    Test Divider
  </span>
  <div
    class="c2"
  />
</div>
`;
